src/gen/src/GLG4VertexGen.cc

```diff
@@ -918,9 +918,13 @@ GeneratePrimaryVertex(G4Event *argEvent, G4ThreeVector &dx, G4double dt)
 				 PHEP3*energy_unit );
     }
     if (PHEP5 != req_novalue)
-      particle->SetMass( PHEP5*energy_unit );
+      particle->SetMass( PHEP5*energy_unit ); //this is not working and I don't know why (F.Sutanto, Apr 2020)
     if (polx != 0.0 || poly != 0.0 || polz != 0.0)
       particle->SetPolarization(polx, poly, polz);
+      
+      //I will just set the mass here (F. Sutanto, Apr 2020)
+      G4ParticleDefinition * myPar = G4ParticleTable::GetParticleTable()->FindParticle(  IDHEP );
+      particle->SetMass( myPar->GetPDGMass() );
```

Comment:
It works until early 2017ish (with geant 10.3). I am not sure when it stops working.

req_novalue is set at 1e30 GeV. If the user inputs a different value, then 1e30 is changed into whatever value that user inputs. 

PHEP5 is not necessary at all (redundant information).... I am guessing that the original writer just copy paste the definition of G4HEPEvtParticle in geant4...
Since the user of external generator provides the pdg number, Geant4 should be able to find its mass from the particle table. 

I would actually prefer to delete line 921 since it's not necessary.
But I decided to just overwrite it in case if one day line 921 will work again (perhaps with different version of geant4?)